atmcfg/atmcfg.go

```diff
@@ -55,6 +56,32 @@ func AddUser(out *opsmngr.AutomationConfig, u *opsmngr.MongoDBUser) {
 	out.Auth.Users = append(out.Auth.Users, u)
 }
 
+// AddIndexConfig adds an IndexConfig to the authentication config
+func AddIndexConfig(out *opsmngr.AutomationConfig, newIndex *opsmngr.IndexConfigs) error {
+	if out == nil {
+		return errors.New("the Automation Config has not been initialized")
+	}
+	_, exists := search.MongoDBIndexes(out.IndexConfigs, func(index *opsmngr.IndexConfigs) bool {
+		if index.RSName == newIndex.RSName && index.CollectionName == newIndex.CollectionName && index.DBName == newIndex.DBName && len(index.Key) == len(newIndex.Key) {
+			// if keys are the equal the two indexes are considered to be the same
+			for i := 0; i < len(index.Key); i++ {
+				if index.Key[i][0] != newIndex.Key[i][0] || index.Key[i][1] != newIndex.Key[i][1] {
+					return false
+				}
+			}
+
+			return true
+		}
+
+		return false
+	})
```

Comment:
I am not sure how to implement this. 
Inside that function I am using `newIndex` which is a parameter of `AddIndexConfig`. If I pass `newIndex` in  `mySearch` then I would have to change `search.MongoDBIndexes` as well since the `f` is taking more than 1 param and I am not sure how to change it.


```
func MongoDBIndexes(a []*opsmngr.IndexConfigs, f func(configs *opsmngr.IndexConfigs) bool) (int, bool) {
	for i, m := range a {
		if f(m) {
			return i, true
		}
	}
	return len(a), false
}`

Reply:
The only way that I see is to pass the `newIndex` to the method `MongoDBIndexes`

Reply:
you can decorate it
```go
func mySearch(newIndex *opsmngr.IndexConfigs) func(index *opsmngr.IndexConfigs) bool { ...}
  return func(index *opsmngr.IndexConfigs) bool {...}
}
```
A func that returns a func ;)